src/topIssues/topIssues.service.ts

```diff
@@ -0,0 +1,83 @@
+import { BadRequestException, Injectable, InternalServerErrorException, Logger, NotFoundException } from '@nestjs/common';
+import { CreateTopIssueDto, TopIssueOutputDto, UpdateTopIssueDto } from './schemas/topIssues.schema';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { TopIssue } from '@prisma/client';
+import { PrismaClientValidationError } from '@prisma/client/runtime/library';
+
+@Injectable()
+export class TopIssuesService {
+  private readonly logger = new Logger(TopIssuesService.name);
+  constructor(private prismaService: PrismaService) {}
+
+  async create(body: CreateTopIssueDto): Promise<TopIssueOutputDto> {
+    const { name, icon } = body;
+    return await this.prismaService.topIssue.create({
+      data: {
+        name, 
+        icon
+      }
+    })
+  }
+
+  async update(id: number, body: UpdateTopIssueDto): Promise<TopIssue> {
+    const { name, icon } = body;
+      return await this.prismaService.topIssue.update({
+        where: { id },
+        data: { name, icon }
+      });
+  }
+
+  async delete(id: number): Promise<void> {
+    const issue = await this.prismaService.topIssue.findUnique({
+      where: { id },
+      include: {
+        positions: true,
+        campaignPositions: true,
+        campaigns: true,
+      },
+    });
+
+    if (!issue) {
+      this.logger.error(`Failed to delete Top Issue with id ${id}`);
+      throw new NotFoundException(`Top issue with id ${id} not found`);
+    }
+
+    const positionIds = issue.positions.map((position) => position.id);
+    const campaignPositionIds = issue.campaignPositions.map((cp) => cp.id);
+    
+
+    await this.prismaService.$transaction(async (prisma) => {
+      if (campaignPositionIds.length > 0) {
+        await prisma.campaignPosition.deleteMany({
+          where: { id: { in: campaignPositionIds } },
+        });
+      }
+
+      if (positionIds.length > 0) {
+        await prisma.position.deleteMany({
+          where: { id: { in: positionIds } },
+        });
+      }
```

Comment:
Do we still need all of this if the table definition includes `CASCADE DELETE` now? 🤔